### ISSUE TITLE
network doesn't generalize well

### ISSUE THREAD

**Issue by ysnylmzz** [polimi-ispl/icpr2020dfdc#62]
Hi, I tried a few samples from faceshifter paper ( https://arxiv.org/pdf/1912.13457.pdf ) and also tried on obama's deepfake video ( https://www.youtube.com/watch?v=cQ54GDm1eL0 ) . Could Not detect

**Comment by CrohnEngineer** [comment]
Hey @ysnylmzz ,

thanks for passing by and checking our code!
We are aware of performance degradations of our method on datasets and face manipulation methods not seen during training.
Indeed, as long as I know, FaceShifter has been added to FaceForensics only recently, and definitely was not present at the time we have written and presented the paper (and also trained the models).
In order to improve the detection performances also on FaceShifter samples, a naive yet effective approach could be to finetune the models we provide on such videos. 
We have also written a [paper](https://ieeexplore.ieee.org/abstract/document/9360901) on the best practices we've found to be useful for generalizing across different datasets if you are interested in the topic.
Regarding the Obama video, did you download it directly from Youtube?
If that is the case, unfortunately social media sharing of videos use severe compression algorithms that often wipe out useful forensic traces: that might explain why you couldn't detect the "fakeness" of it.
The deployment of academic solutions like ours in real-world scenario like this is still an open research topic.
If you have any other question just let us know!
Bests,

Edoardo

**Comment by yasinylmzz** [comment]
Thank you for your answer. Yes, I got the video from youtube. It may have happened as you said. I tried it on a few deepfake videos as well. It comes in around 0.3 on average. In real videos, it is almost 0. Although the network has not been fully generalized, it is still seen that this information is available in the features. we agree on the difficulty of real-world scenarios. Congratulations on your paper

Best regards

Yasin
